### DISCUSSION TITLE
ImgComponent as Image from next.js

### DISCUSSION THREAD

**Discussion by uigywnkiub** [heroui-inc/heroui#3295] [Help] [answered]
How can I use Image component from next.js as props ImgComponent in User component from nextui?
![image](https://github.com/nextui-org/nextui/assets/29861553/1f49e299-961f-4fe3-a574-25d9aa38a399)

**Comment by wingkwong** [comment]
You can do in this way.

```tsx
avatarProps={{
  ImgComponent: Image,
  imgProps: {
    src: session?.user.image || '',
    width: 40,
    height: 40,
    alt: 'U',
  },
}}
```

**Reply by uigywnkiub** [reply]
> You can do in this way.
> 
> ```tsx
> avatarProps={{
>   ImgComponent: Image,
>   imgProps: {
>     src: session?.user.image || '',
>     width: 40,
>     height: 40,
>     alt: 'U',
>   },
> }}
> ```

Photo not appear and img component too. And invalid props from Image from next... Looks like it didn't work, because I tried to use plain img as ImpComponent and it isn’t as img in DOM
<img width="1475" alt="image" src="https://github.com/nextui-org/nextui/assets/29861553/82e69dc5-f75f-4360-a45c-d7c0aa60c229">

**Reply by wingkwong** [reply, marked as answer]
Can you try the following - putting src and alt outside imgProps.

```tsx
avatarProps={{
  ImgComponent: NextImage,
  imgProps: {
    width: 40,
    height: 40,
  },
  src: session?.user.image || '',
  alt: "U",
}}
```

**Reply by uigywnkiub** [reply]
> Can you try the following - putting src and alt outside imgProps.
> 
> ```tsx
> avatarProps={{
>   ImgComponent: NextImage,
>   imgProps: {
>     width: 40,
>     height: 40,
>   },
>   src: session?.user.image || '',
>   alt: "U",
> }}
> ```

Works, thanks! 
In the future, it will be nice to type hints from next.js image for instance. And for instance props like placeholder='blur', not applied...
```js
imgProps: {
          // @ts-ignore
          priority: true,
          quality: 100,
          width: 40,
          height: 40,
        },
```
